\documentclass[12pt,twoside, lineno]{gsajnl}
% Use the documentclass option 'lineno' to view line numbers
\usepackage{tabularx}
\usepackage{longtable}
\articletype{mp} % article type

\title{Testing pleiotropy vs.\ separate QTL in multiparental populations}

\author[$\ast$,1]{Frederick J. Boehm}
\author[$\dagger$]{Elissa J. Chesler}
\author[$\ast$, $\ddagger$]{Brian S. Yandell}
\author[$\S$]{Karl W. Broman}

\affil[$\ast$]{Department of Statistics, University of Wisconsin-Madison, Madison, Wisconsin 53706}
\affil[$\dagger$]{The Jackson Laboratory, Bar Harbor, Maine 04609}
\affil[$\ddagger$]{Department of Horticulture, University of Wisconsin-Madison, Madison, Wisconsin 53706}
\affil[$\S$]{Department of Biostatistics and Medical Informatics, University of Wisconsin-Madison, Madison, Wisconsin 53706}

\keywords{Quantitative trait locus; pleiotropy; multivariate analysis; linear mixed effects models; systems genetics; ...}

\runningtitle{Testing pleiotropy in MPPs} % For use in the footer
\runningauthor{Boehm \textit{et al.}}




\begin{abstract}

The high mapping resolution of multiparental populations, combined
with technology to measure tens of thousands of phenotypes, presents
an opportunity for quantitative methods to enhance understanding of
the genetic architecture of complex traits. When multiple traits map
to a common genomic region, knowledge of the number of distinct loci
provides important insight into the underlying mechanism and can
assist planning for subsequent experiments. We extend the work of
\citet{jiang1995multiple} to develop a likelihood ratio test, for a
pair of traits, of the null hypothesis of pleiotropy against the
alternative hypothesis of separate QTL, for the case of more than two
alleles. We also incorporate polygenic random effects to account for
variable relatedness among subjects. We use a parametric bootstrap to
determine statistical significance. We apply our methods to a
behavioral genetics data set from Diversity Outbred mice, where we
find evidence for presence of two distinct loci in a 2.5~cM region.
Our methods have been incorporated into the R package
package \texttt{qtl2pleio}.

\end{abstract}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Stuff between the two lines of percentage marks may need to be removed before the draft is finalized.
\usepackage[colorinlistoftodos, prependcaption,textsize=tiny]{todonotes}
%% use 'disable' option for todonotes to disable the package, as in the commented line below.
% \usepackage[disable, colorinlistoftodos, prependcaption,textsize=tiny]{todonotes}
\presetkeys%
    {todonotes}%
    {inline,backgroundcolor=yellow}{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\setboolean{displaycopyright}{true}

\begin{document}

\maketitle
\thispagestyle{firststyle}
\marginmark
\firstpagefootnote
\correspondingauthoraffiliation{Department of Statistics, 1220 Medical Sciences Center, 1300 University Avenue, Madison, WI 53706. E-mail: frederick.boehm@gmail.com}
\vspace{-11pt}%



Complex trait studies in multiparental populations present new
challenges in statistical methods and data analysis. Among these is
the development of strategies for multivariate trait analysis. The
joint analysis of two or more traits allows one to address additional
questions, such as whether two traits share a single pleiotropic
locus.




Previous research addressed the question of pleiotropy vs.\ separate
QTL in two-parent crosses.
\citet{jiang1995multiple} developed a likelihood
ratio test for pleiotropy vs.\ separate QTL for a pair of traits.
Their approach assumed that each trait was affected by a single QTL.
Under the null hypothesis, the two traits were affected by a common
QTL, and under the alternative hypothesis the two traits were affected
by distinct QTL.
\citet{knott2000multitrait} used linear regression to develop a fast
approximation to the test of \citet{jiang1995multiple}, while
\citet{tian2016dissection} used the methods from
\citet{knott2000multitrait} to dissect QTL hotspots in a F$_2$
population.


Multiparental populations, such
as the Diversity Outbred (DO)mouse population \citep{churchill2012diversity}, enable high-precision
mapping of complex traits \citep{de2014genetics}. The DO
mouse population began with progenitors of the Collaborative
Cross (CC) mice \citep{churchill2004collaborative}
Each DO mouse is a highly heterozygous genetic mosaic
of alleles from the eight CC founder lines. Random
matings among non-siblings have maintained the DO
population for more than 23 generations \citep{chesler2016diversity}.

Several limitations of previous pleiotropy vs.\ separate QTL tests
prevent their direct application in multiparental populations. First,
multiparental populations can have complex patterns of relatedness
among subjects, and failure to account for these patterns of
relatedness may lead to spurious results \citep{yang2014advantages}.
Second, previous tests allowed for only two founder lines
\citep{jiang1995multiple}. Finally, \citet{jiang1995multiple} assumed
that the null distribution of the test statistic follows a chi-square
distribution.

We developed a pleiotropy vs.\ separate QTL test for two traits in
multiparental populations. Our test builds on research that
\citet{jiang1995multiple}, \citet{knott2000multitrait},
\citet{tian2016dissection}, and \citet{zhou2014efficient} initiated.
Our innovations include the accommodation of $k$ founder alleles per
locus (compared to the traditional two founder alleles per locus) and
the incorporation of multivariate polygenic random effects to account
for relatedness. Furthermore, we implemented a parametric bootstrap to
calibrate test statistic values \citep{efron1979,tian2016dissection}.

Below, we describe our likelihood ratio test for pleiotropy vs.\
separate QTL. In simulation studies, we find that it is slightly
conservative, and that it has power to detect two separate loci when
the univariate LOD peaks are strong. We further illustrate our
approach with an application to data on a pair of behavior traits in
a population of 261 DO mice \citep{logan2013high,recla2014precise}.
We find modest evidence for distinct QTL in a 2.5-cM region on mouse
chromosome 8.


\section{Methods}
\label{sec:materials:methods}

Our strategy involves first identifying two traits that map to a common
genomic region. We then perform a two-dimensional, two-QTL scan over
the genomic region, with each trait affected by one QTL of varying
position. We identify the QTL position that maximizes the likelihood
under pleiotropy (that is, along the diagonal where the two QTL are at
a common location), and the ordered pair of positions that maximizes
the likelihood under the model where the two QTL are allowed to be
distinct. The logarithm of the ratio of the two likelihoods is our
test statistic. We calibrate this test statistic with a parametric
bootstrap.

\subsection{Data structures}

The data consist of three objects. The first is an $n$ by $k$ by $m$
array of allele probabilities for $n$ subjects with $k$ alleles and
$m$ marker positions on a single chromosome [derived from the observed
SNP genotype data by a hidden Markov model; see
\citet{broman2019rqtl2}]. The second object is an $n$ by 2 matrix of
phenotype values. Each column is a phenotype and each row is a
subject. The third object is an $n$ by $c$ matrix of covariates, where
each row is a subject and each column is a covariate.

One additional object is the genotype-derived kinship matrix, which is
used in the linear mixed model to account for population structure. We
are focusing on a defined genomic interval, and we prefer to use a
kinship matrix derived by the ``leave one chromosome out'' (LOCO)
method \citep{yang2014advantages}, in which the kinship matrix is
derived from the genotypes for all chromosomes except the chromosome
under test.




\subsection{Statistical Models}

Focusing on a pair of traits and a particular genomic region of
interest, the next step is a two-dimensional, two-QTL
scan \citep{jiang1995multiple}. We consider two QTL with each
affecting a different trait, and consider all possible pairs of
locations for the two QTL. For each pair of positions, we fit
the multivariate linear mixed effects model defined in Equation
\ref{eqn:model1}. Note that we have
assumed an additive genetic model throughout our analyses, but
extensions to design matrices that include dominance are
straightforward.


\begin{equation}
vec(Y) = X vec(B) + vec(G) + vec(E)
\label{eqn:model1}
\end{equation}
where $Y$ is the $n$ by $2$ matrix of phenotypes values;
$X$ is a $2n$ by $2(k + c)$
matrix that contains the $k$ allele probabilities for the two QTL
positions and the $c$
covariates in diagonal blocks; $B$ is a $(k + c)$ by $2$ matrix of
allele effects and covariate effects; $G$ is a $n$ by $2$ matrix of
random effects; and $E$ is a $n$ by $2$ matrix of random errors. $n$
is the number of mice. The `vec' operator stacks columns from a matrix
into a single vector. For example, a 2 by 2 matrix inputted to `vec'
results in a vector with length 4. Its first two entries are the
matrix's first column, while the third and fourth entries are the
matrix's second column.


We also impose distributional assumptions on $G$ and $E$:

\begin{equation}
G \sim MN_{n x 2}(0, K, V_g)
\label{eqn:model2}
\end{equation}

and

\begin{equation}
E \sim MN_{nx2}(0, I, V_e)
\label{eqn:model3}
\end{equation}
where $MN_{n x 2}(0, V_r, V_c)$ denotes the matrix-variate ($n$ by 2)
normal distribution with mean being the $n$ by $2$ matrix with all
zero entries and row covariance $V_r$ and column covariance $V_c$. We
assume that $G$ and $E$ are independent.


\subsection{Parameter inference and log likelihood calculation}

Inference for parameters in multivariate linear mixed effects models
is notoriously difficult and computationally intense
\citep{meyer1989restricted,meyer1991estimating}. Thus, we make the
simplifying assumption that $V_g$ and $V_e$ don't depend on allele
probabilities. We use restricted maximum likelihood methods to fit the
model:

\begin{equation}
vec(Y) = X_0vec(B) + vec(G) + vec(E)
\label{model}
\end{equation}
$X_0$ is a $2n$ by $2(c + 1)$ matrix. The first column of each
diagonal block $X_0$ has all entries equal to one, and the remaining
columns are covariates.

We draw on our implementation \citep{gemma2} (in \texttt{R}) of the
GEMMA algorithm for fitting a multivariate linear mixed effects model
with expectation-maximization \citep{zhou2014efficient}. We use
restricted maximum likelihood fits for the variance components $V_g$
and $V_e$ in subsequent calculations of the generalized least squares
solution $\hat B$.

\begin{equation}
    \hat B = (X^T\hat\Sigma^{-1}X)^{-1}X^T\hat\Sigma^{-1}vec(Y)
\end{equation}

\noindent where

\begin{equation}
    \hat\Sigma = \hat V_g \otimes K + \hat V_e \otimes I_n
    \label{cov}
\end{equation}

\noindent where $\otimes$ denotes the Kronecker product, $K$ is a
("leave one chromosome out") kinship matrix, and $I_n$ is a n by n
identity matrix. We then calculate the log likelihood for a normal
distribution with mean $X vec(\hat B)$ and covariance $\hat \Sigma$
that depends on our estimates of $V_g$ and $V_e$ (Equation \ref{cov}).

\subsection{Pleiotropy vs.\ separate QTL hypothesis testing framework}

Our test applies to two traits considered simultaneously. Below,
$\omega_1$ and $\omega_2$ denote putative locus positions for traits
one and two. We quantitatively state the competing hypotheses for our
test as:

\begin{eqnarray}
H_0: \omega_1 = \omega_2 \nonumber\\
H_A: \omega_1 \neq \omega_2
\label{eqn:hypotheses}
\end{eqnarray}

\noindent Our likelihood ratio test statistic is:

\begin{equation}
- \log_{10} \frac{\max L_0(B, \Sigma, \omega_1, \omega_2)}{\max L_A(B, \Sigma, \omega_1, \omega_2)}
\label{eqn:test-statistic}
\end{equation}
where $L_0$ is the likelihood under the null hypothesis (pleiotropy)
and $L_A$ is the likelihood under the alternative hypothesis (separate
QTL). We maximize each term in the fraction over its respective
parameter space.

\subsection{Visualizing profile LOD traces}

Following an innovation of \citet{zeng2000genetic} and
\citet{tian2016dissection}, we plotted three traces: profile LOD$_1$
trace, profile LOD$_2$ trace, and the pleiotropy trace to aid in
visualizing the two-dimensional likelihood surface. By definition
(Equation \ref{eq:lodp}), the maximum height of the pleiotropy trace
is zero. Additionally, the maxima for profile LOD$_1$ trace and
profile LOD$_2$ trace are equal and non-negative (Equations
\ref{eq:lodpvl} and \ref{eq:profilelod}).

We define the $LOD$ for our test:

\begin{equation}
LOD(\omega_1, \omega_2) = ll_{10}(\omega_1, \omega_2) - \max ll_{10}(\omega, \omega)
\label{eq:lodpvl}
\end{equation}
where $ll_{10}$ denotes log-likelihood.

We follow \citet{zeng2000genetic} and \citet{tian2016dissection} in
defining profile LOD by the equation

\begin{equation}
\text{profile LOD}_1(\omega_1) = \max_{\omega_2}\text{LOD}_{pvl}(\omega_1, \omega_2)
\label{eq:profilelod}
\end{equation}
We define profile LOD$_2(\omega_2)$ analogously.

We constructed the pleiotropy trace by calculating the log-likelihoods
for the pleiotropic models at every marker.

\begin{equation}
LOD_{p}(\omega) = ll_{10}(\omega, \omega) - \max ll_{10}(\omega, \omega)
\label{eq:lodp}
\end{equation}




\subsection{Bootstrap for test statistic calibration}

We used a parametric bootstrap to calibrate our test statistic
\citep{efron1979}. While \citet{jiang1995multiple} used quantiles of a
chi-squared distribution to determine p-values, apparent deviations of
the test statistic distribution from a chi-squared concerned us. Thus,
we followed the approach of \citet{tian2016dissection} by identifying
the marker that maximizes the likelihood under pleiotropy (the null
hypothesis). We then used the inferred model parameters (under
pleiotropy) at this marker to create bootstrap data sets, according to
equations \ref{eqn:model1}, \ref{eqn:model2}, and \ref{eqn:model3}.
Specifically, we created $b$ bootstrap data sets for each analysis. We
performed a two-dimensional QTL scan (over the genomic region of
interest) with each of the $b$ bootstrap samples to get $b$ test
statistic values. We treated these $b$ test statistics as the
empirical distribution with which we compared the test statistic for
the observed data. We calculated the p-value as the proportion of the
$b$ test statistics ($\Lambda_i$) that equaled or exceeded the test
statistic value for the true data, $\Lambda$:

\begin{equation}
p = \frac{\# \lbrace i:\Lambda_i \geq \Lambda\rbrace}{b}
\end{equation}
where $\Lambda_{i}$ denotes the likelihood ratio test statistic for
the $i^{th}$ bootstrap data set and $\Lambda$ denotes the true
likelihood ratio test statistic, \textit{i.e.}, that for the observed
data.

\subsection{Data \& Software Availability}

Our methods have been implemented in an R package, \texttt{qtl2pleio},
available at GitHub:

\href{https://github.com/fboehm/qtl2pleio}{https://github.com/fboehm/qtl2pleio}

\noindent Custom R code for our analyses and simulations are at GitHub:

\href{https://github.com/fboehm/qtl2pleio-manuscript}{https://github.com/fboehm/qt2pleio-manuscript}

\noindent The data from \citet{recla2014precise} and
\citet{logan2013high} are available at the Mouse Phenome Database:

\href{https://phenome.jax.org/projects/Chesler4}{https://phenome.jax.org/projects/Chesler4} and \href{https://phenome.jax.org/projects/Recla1}{https://phenome.jax.org/projects/Recla1}.

\noindent They are also available in R/qtl2 format at
\href{https://github.com/rqtl/qtl2data}{https://github.com/rqtl/qtl2data}.




\section{Simulation studies}

We performed two types of simulation studies, one for type I error
rate assessment and one to characterize statistical power to detect
separate QTL. To simulate traits, we specified $X$, $B$, $V_g$, $K$,
and $V_e$ matrices (Equations \ref{eqn:model1}, \ref{eqn:model2},
\ref{eqn:model3}). For all type I error rate studies, we used the
allele probabilities from a single locus for 479 Diversity Outbred
mice. For studies to assess power to detect separate QTL, we used the
same 479 Diversity Outbred mice and allele probabilities of markers
that are centered at the marker fro the type I error rate studies.

\subsection{Type I error rate analysis}

To quantify type I error rate ({\em i.e.}, false positive rate), we
simulated 400 bivariate traits for each of eight sets of parameter
inputs (Table \ref{table-typeI}). We used a $2^3$ factorial
experimental design with three factors: allele effects difference,
allele effects partitioning, and genetic correlation, \textit{i.e.},
the off-diagonal entry in the 2 by 2 matrix $V_g$.

\begin{center}
\begin{table}
\small
  \begin{tabular}{ c | c | c | c | c}
    \hline
    Run & $\Delta$(Allele effects) & Allele effects partitioning & Genetic correlation & Type I error rate \\ \hline
    1 & 6 & ABCD:EFGH & 0 & 0.0325\\
    2 & 6 & ABCD:EFGH & 0.6 & 0.035\\
    3 & 6 & F:ABCDEGH & 0 & 0.040\\
    4 & 6 & F:ABCDEGH & 0.6 & 0.045\\
    5 & 12 & ABCD:EFGH & 0 & 0.0375\\
    6 & 12 & ABCD:EFGH & 0.6 & 0.0425\\
    7 & 12 & F:ABCDEGH & 0 & 0.025\\
    8 & 12 & F:ABCDEGH & 0.6 & 0.025\\
    \hline
  \end{tabular}
  \caption{Type I error rates for all eight runs in our $2^3$
    experimental design. We set (marginal) genetic variances,
    \emph{i.e.}, diagonal elements of $V_g$, to 1 in all runs. $V_e$
    was set to the 2 by 2 identity matrix in all runs. We used allele
    probabilities at a single genetic marker to simulate traits for
    all eight sets of parameter inputs. Founder allele one-letter
    abbreviations are listed in Table \ref{table-letters}. In the
    column "Allele effects partitioning", "ABCD:EFGH" means that lines
    A, B, C, and D form one group while the other four lines are the
    second group. "F:ABCDEGH" means that the F allele is one group,
    while the seven other lines are a second group. Thus, for run 1,
    we assigned lines A, B, C, and D to have founder allele effects of
    $3$, while E, F, G, and H had founder allele effects of $-3$.}
  \label{table-typeI}
  \end{table}
\end{center}

We explain below the reasoning behind our choice of levels for the
three factors. We chose two strong allele effects difference values, 6
and 12. These ensured that the univariate phenotypes mapped with high
LOD scores to the region of interest. For one level of the allele
partitioning factor, we isolated the CAST allele (F). Three
considerations motivated this choice. First, we observed in other
analyses that CAST allele effects often differed from those of the
other seven founder lines. Second, the phylogenetic distance between
CAST and the other lines is greater than other inter-line distances
\citep{didion2013deconstructing}. Third, we wanted to examine the
effect of sample sizes (approximately equal vs unequal) in our
analyses. We chose to evenly split the founder alleles into two groups
of four for the other level of the allele partitioning factor (Table
\ref{table-typeI}). We considered two values, 0 and 0.6, of the
genetic correlation, \textit{i.e.}, the off-diagonal entry in $V_g$.
The marginal genetic variances (\textit{i.e.}, the diagonal entries in
$V_g$) for each trait were always set to one.

We performed 400 tests per set of parameter inputs. Each test used $b
= 400$ bootstrap samples. For each bootstrap sample, we calculated the
test statistic (Equation \ref{eqn:test-statistic}). We then compared
the test statistic from the simulated trait against the empirical
distribution of its 400 bootstrap test statistics. When the simulated
trait's test statistic exceeded the 0.95 quantile of the empirical
distribution of bootstrap test statistics, we rejected the null
hypothesis. We observed that the test is slightly conservative over
our range of parameter selections (Table \ref{table-typeI}).


\subsection{Power analysis}

We also investigated the test's power to detect the presence of two
distinct QTL. We used a 2 by 2 by 5 experimental design, where our
three factors were allele effects difference, allele effects
partitioning, and inter-locus distance. The two levels of allele
effects difference were 1 and 2. The two levels of allele effects
partitioning were, as in the type I error rate studies, ABCD:EFGH and
F:ABCDEGH (Table \ref{table-letters}). The five levels of interlocus
distance were 0, 0.5, 1, 2, and 3 cM. $V_g$ and $V_e$ were both set to
the 2 by 2 identity matrix in all power study simulations.

We simulated 400 bivariate traits per set of parameter inputs. For
each simulated bivariate trait, we calculated a likelihood ratio test
statistic. We then applied our parametric bootstrap to calibrate the
test statistics. For each bivariate trait, we used $b = 400$ bootstrap
samples. Because the bootstrap test statistics within a single set of
parameter inputs followed approximately the same distribution, we
pooled the $400 * 400 = 160,000$ bootstrap samples per set of
parameter inputs and compared each test statistic to the empirical
distribution derived from the 160,000 bootstrap samples. However, for
parameter inputs with interlocus distance equal to zero, we didn't
pool the 160,000 bootstrap samples; instead, we proceeded by
calculating power, \textit{i.e.}, type I error rate, as we did in the
type I error rate study above.

\begin{figure}
\includegraphics[width = \textwidth]{../R/power-curves.eps}
\caption{Pleiotropy vs.\ separate QTL power curves for each of four
  sets of parameter settings. Factors that differ among the four
  curves are allele effects difference and allele partitioning. Olive
  green denotes high (2) allele effects difference and allele
  partitioning into two groups of four. Purple denotes high (2) allele
  effects difference with unequal allele partitioning. Red and blue
  both have the low (1) allele effects difference with even and
  uneven, respectively, allele partitioning.}
\label{fig:power}
\end{figure}

We present our power study results in Figure \ref{fig:power}. Each
curve tends upward monotonically as interlocus distance increases. The
curve with both the high value, 2, of allele effects difference and
even allele partitioning into two groups of four founder alleles
uniformly has the greatest power (olive green in Figure
\ref{fig:power}). The purple curve denotes the parameter input sets
with uneven allele partitioning (F:ABCDEGH) and the high value (2) of
allele effects difference. The two curves - red and blue - with the
low value, 1, of allele effects difference are uniformly less powerful
than those with the high value of allele effects difference. Within
the two curves with allele effects difference equal to one, that with
the even partitioning (ABCD:EFGH) of founder alleles has uniformly
greater power.


\section{Application}
\label{sec:app}

To illustrate our methods, we applied our test to data from
\citet{logan2013high} and \citet{recla2014precise}.
\citet{logan2013high} identified two phenotypes - "percent time in
light" and "distance traveled in light" - that mapped to 55 cM on
chromosome 8. This genomic region intrigued us because
\citet{recla2014precise} identified \textit{Hydin} as the gene that
underlies the mapping of "hot plate latency" to 57 cM (Supplementary
figures). This led us to ask whether \textit{Hydin} affects "percent
time in light". We ignored "distance traveled in light" because of its
high correlation (Pearson correlation 0.89) with "percent time in
light".




We downloaded and formatted the data from the Mouse Phenome Database
\citep{bogue2015collaborative}. We inserted pseudomarkers at 0.1 cM
intervals to bring the number of genome-wide markers to 20,183. We
used three data structures: one for genotype data, one for phenotype
data, and a third for covariates data. For each mouse and each marker,
we have eight founder allele probabilities. We structured the allele
probabilities for each chromosome as a 261 by 8 by $m_i$
three-dimensional array, where $m_i$ denotes the number of markers on
chromosome $i$. We arranged phenotype data into a 261 by 3 matrix,
where each column was a single phenotype and each row corresponded to
a single mouse. We used only the sex covariate in our analyses.

We briefly describe the phenotypes that \citet{logan2013high} and
\citet{recla2014precise} measured. Both "percent time in light" and
"distance traveled in light" were measured by placing mice in a
light-dark box, consisting of an insert dividing an open-field
apparatus into a light component and a dark component.
\citet{logan2013high} then measured the percentage of 20 minutes time
spent in the light component of the light-dark box. "Hot plate
latency" reflects pain reflexes in response to a thermal stimulus, as
recorded by a hot plate analgesia meter. Each mouse remained on the
plate until it performed either of two behaviors regarded as
indicative of nociception: hindpaw lick or hindpaw shake/flutter.




To determine whether \textit{Hydin} affects "percent time in light",
we examined a phenotype scatter plot, performed a test of pleiotropy
vs.\ separate QTL, and created founder allele effects plots. A scatter
plot revealed that "percent time in light" and "hot plate latency" are
not highly correlated, with a Pearson correlation coefficient of
$-0.15$ (Figure \ref{fig:scatter}).

\begin{figure}
\includegraphics[width = \textwidth]{../Rmd/scatter.eps}
\caption{Scatter plot of "hot plate latency" against "percent time in
  light", after applying logarithm transformations and winsorizing
  both traits.}
\label{fig:scatter}
\end{figure}


\begin{figure}
\includegraphics[width = \textwidth]{../Rmd/chr8-lods.eps}
\caption{Chromosome 8 univariate LOD scores for percent time in light
  and hot plate latency reveal broad, overlapping peaks between 53 cM
  and 64 cM. The peak for percent time in light spans the region from
  approximately 53 cM to 60 cM, with a maximum near 55 cM. The peak
  for hot plate latency begins near 56 cM and ends about 64 cM.}
\label{fig:chr8-lod}
\end{figure}



In examining the founder allele effects plot for "percent time in
light", we see that the purple (WSB) and red (PWK) alleles have high
effects near 55 cM, while the navy blue (NOD) has a dramatically lower
value (Figure \ref{fig:chr8-effects}). The allele effects plot for
"hot plate latency", on the other hand, features a noticeably distinct
pattern of allele effects near 55 to 60 cM (Figure
\ref{fig:chr8-effects}). Specifically, it has grey (B6) and green
(CAST) alleles with much lower effects, while the pink (129), yellow
(AJ), purple (WSB), and light blue (NZO) all show higher allele
effects in this region.

\begin{figure}
\includegraphics[width = \textwidth]{../Rmd/coefs.eps}
\caption{Chromosome 8 univariate LOD scores for percent time in light
  and hot plate latency reveal broad, overlapping peaks between 53 cM
  and 64 cM. The peak for percent time in light spans the region from
  approximately 53 cM to 60 cM, with a maximum near 55 cM. The peak
  for hot plate latency begins near 56 cM and ends about 64 cM.}
\label{fig:chr8-effects}
\end{figure}




We then performed a two-dimensional QTL scan for the pair of
phenotypes. With these results, we created a profile LOD plot (Figure
\ref{fig:profiles}). The profile LOD plot for the analysis of "percent
time in light" and "hot plate latency" displays a peak in the "percent
time in light" profile LOD trace near 55 cM; in fact, it overlaps
perfectly with the univariate LOD peak for "percent time in light".
"Hot plate latency", on the other hand, demonstrates a broad peak in
the profile LOD trace, with the maximum achieved near 57 cM (Figure
\ref{fig:profiles}A). The pleiotropy trace maximum occurs, too, near
57 cM. The maximal heights of the two profile LOD traces are near 1.2,
which is the likelihood ratio test statistic value.

Using the two-dimensional QTL scan results, we calculated a likelihood
ratio test statistic value: 1.20. We also calculated a bootstrap
p-value with $b = 1000$. The p-value was 0.11, indicating modest
evidence for distinct QTL for the two traits.


\begin{figure}
\includegraphics[width = \textwidth]{../Rmd/profile.eps}
\caption{Profile LOD curves for the pleiotropy vs.\ separate QTL
  hypothesis test for "percent time in light" and "hot plate latency".
  Gray trace denotes pleiotropy LOD values. Triangles denote the
  univariate LOD maxima, while diamonds denote the profile LOD maxima.
  For "percent time in light", the yellow triangle obscures the
  smaller yellow diamond. Likelihood ratio test statistic value
  corresponds to the height of the blue and yellow traces at their
  maxima.}
\label{fig:profiles}
\end{figure}








\section{Discussion}

We developed a test of pleiotropy vs.\ separate QTL for multiparental
populations, extending the work of \citet{jiang1995multiple} for
multiple alleles and with a linear mixed models to account for
population structure. Our simulation
studies indicate that the test has power to detect presence of
separate loci, especially when univariate trait associations are
strong (Figure \ref{fig:power}). Type I error rates indicate that our
test is slightly conservative (Table \ref{table-typeI}).

In the application of our method to two behavioral phenotypes in a
study of 261 Diversity Outbred mice
\citep{recla2014precise,logan2013high}, we obtained weak evidence
(p=0.11) for the presence of two distinct QTL, with one QTL (which
contained \textit{Hydin}) affecting only "hot plate latency" and a
second QTL affecting "percent time in light" (Figure
\ref{fig:profiles}).

Founder allele effects plots provide further evidence for the presence
of two distinct loci. As \citet{macdonald2007joint} and
\citet{king2012genetic} argue in their analyses of multiparental
*Drosophila* populations, a biallelic pleiotropic QTL would result in
allele effects plots that have similar patterns. While we don't know
that "percent time in light" and "hot plate latency" arise from
biallelic QTL, the dramatic differences that we observe in allele
effects patterns further support the argument for two distinct loci,
with *Hydin* affecting only "hot plate latency".

Before discussing future directions for our research, we consider
limitations of our current software and methods. Analyses with
\texttt{qtl2pleio} are slow. One factor contributing to this is the
fact that we wrote \texttt{qtl2pleio} mostly in R. R, due to its
design, is slower than many compiled languages, such as C++
\citep{wickham2014advanced}. To accelerate our multi-dimensional QTL
scans, we integrated C++ code into \texttt{qtl2pleio}
\citep{eddelbuettel2011rcpp}.

Another computational bottleneck is the estimation of variance
components $V_g$ and $V_e$. In future research, where we may develop
tests for simultaneous analysis of more than two traits, we will
consider other strategies for variance component estimation, including
that described by \citet{hannah2018limmbo}, who implement a bootstrap
strategy to estimate variance components for lower-dimensional
phenotypes before combining bootstrap variance component estimates
into valid covariance matrices for the full multivariate phenotype.
We'll add this variance component estimation method to
\texttt{qtl2pleio} by using the \texttt{reticulate} R package to
expose objects from the python module \texttt{limmbo}
\citep{reticulate}.


We now contrast pleiotropy vs.\ separate QTL tests with mediation tests
in efforts to illustrate their complementary roles. Many systems
genetics investigations apply mediation tests and related methods to
infer biomolecular causal relationships
\citep{chick2016defining,schadt2005integrative,baron1986moderator}. A
mediation test proceeds by including a putative mediator as a
covariate in the regression analysis of phenotype and QTL. If a
putative mediator sufficiently diminishes the LOD score, then the
putative mediator is declared a mediator of the phenotype-QTL
relationship.

An important distinction lies in the different goals for mediation
tests and pleiotropy vs.\ separate QTL tests. In a mediation test, we
seek to identify mediators of a phenotype-QTL relationship from a
collection of putative mediators. In a pleiotropy vs.\ separate QTL
test, one examines two (or more) phenotypes and inquires about the
number of distinct QTL. A pleiotropy vs.\ separate QTL test provides
information about the number of underlying QTL, while a mediation test
offers conclusions about the relative ordering of biological
intermediates within a molecular pathway.

Despite their distinct goals, \citet{schadt2005integrative} argue that
both a pleiotropy vs.\ separate QTL test and causal inference methods,
such as mediation tests, may contribute to gene network
reconstruction. They develop a model selection strategy, based on
Akaike information criterion \citep{akaike1974new}, to determine which
causal model is most compatible with the observed multivariate
phenotype data \citep{schadt2005integrative}. In this context,
\citet{schadt2005integrative} extend the methods of
\citet{jiang1995multiple} to consider more complicated alternative
hypotheses, such as the possibility of two QTL, one of which
associates with both traits, and one of which associates with only one
trait. As envisioned by \citet{schadt2005integrative}, we foresee
complementary roles emerging for our pleiotropy vs.\ separate QTL test
and mediation tests in the dissection of complex trait genetic
architecture.

Two trends make our scientific contributions especially timely: the
accelerating ability of biologists to measure tens of thousands of
traits in model organisms and the proliferation of model organism
multiparental populations in diverse species. Technological advances
in mass spectrometry and RNA sequencing enable the acquisition of
high-dimensional biomolecular phenotypes
\citep{ozsolak2011rna,han2012multi}. Multiparental populations in
\textit{Arabidopsis}, maize, wheat, oil palm, rice,
\textit{Drosophila}, yeast, and other organisms enable high-precision
QTL mapping \citep{yu2008genetic, tisne2017identification,
  stanley2017genetic, raghavan2017approaches, mackay2012drosophila,
  kover2009multiparent, cubillos2013high}. The need to analyze
high-dimensional phenotypes in multiparental populations compels the
scientific community to develop tools to study genotype-phenotype
relationships and complex trait architecture. Our test, and its future
extensions, will contribute to these ongoing efforts.




\subsection*{Acknowledgments}

This research was performed using the compute resources and assistance
of the UW-Madison Center For High Throughput Computing (CHTC) in the
Department of Computer Sciences. The CHTC is supported by UW-Madison,
the Advanced Computing Initiative, the Wisconsin Alumni Research
Foundation, the Wisconsin Institutes for Discovery, and the National
Science Foundation, and is an active member of the Open Science Grid,
which is supported by the National Science Foundation and the U.S.
Department of Energy's Office of Science.

We thank Lindsay Traeger, Julia Kemis, and Rene Welch for critiquing
early versions of the manuscript.

This work was supported in part by National Institutes of Health grant
R01GM070683 (to K.W.B.). \bibliography{research,qtl2pleio-manuscript}

\newpage
\appendix
\section{Supplementary tables}


\begin{center}
\begin{table}
\small
  \begin{tabular}{ c | c }
    \hline
    Founder allele & One-letter abbreviation \\ \hline
    A/J & A \\
    C57BL/6J & B \\
    129S1/SvImJ & C \\
    NOD/ShiLtJ & D\\
    NZO/H1LTJ & E\\
    Cast/EiJ & F\\
    PWK/PhJ & G\\
    WSB/EiJ & H\\
    \hline
  \end{tabular}
  \caption{Eight founder lines and their one-letter abbreviations.}
  \label{table-letters}
  \end{table}

  % latex table generated in R 3.5.1 by xtable 1.8-3 package
% Tue Nov 20 10:28:47 2018
\begin{table}[ht]
\begin{tabular}{l|lrr}
  \hline
lodcolumn & chr & pos & lod \\
   \hline
percent time in light & 8 & 55.28 & 5.27 \\
 hot plate latency & 8 & 57.77 & 6.22 \\
 percent time in light & 9 & 36.70 & 5.42 \\
 hot plate latency & 9 & 46.85 & 5.22 \\
 percent time in light & 11 & 63.39 & 6.46 \\
 hot plate latency & 12 & 43.52 & 5.13 \\
 percent time in light & 15 & 15.24 & 5.67 \\
 hot plate latency & 19 & 47.80 & 5.48 \\
   \hline
\end{tabular}
  \caption{Both "hot plate latency" and "percent time in light"
    demonstrate multiple QTL peaks with LOD scores above 5.}
  \label{table-peaks}
\end{table}



\end{center}







\newpage
\section{Supplementary figures}
\begin{figure}
\includegraphics{../Rmd/genomewide_lod_trait10.eps}
\caption{Genome-wide QTL scan for percent time in light reveals
  multiple QTL, including one on Chromosome 8.}
\label{fig:genomewide10}
\end{figure}

\begin{figure}
\includegraphics{../Rmd/genomewide_lod_trait22.eps}
\caption{Genome-wide QTL scan for hot plate latency reveals multiple
  QTL, including a peak on Chromosome 8.}
\label{fig:genomewide22}
\end{figure}







\end{document}
